pkg/loghttp/series.go

```diff
@@ -18,18 +19,40 @@ func ParseSeriesQuery(r *http.Request) (*logproto.SeriesRequest, error) {
 	}
 
 	xs := r.Form["match"]
+	// Prometheus encodes with `match[]`; we use both for compatibility.
+	ys := r.Form["match[]"]
+
+	deduped := union(xs, ys)
+	sort.Strings(deduped)
```

Comment:
Can I ask why sorting is required?

Reply:
I did it for deterministic tests, but you're totally right - I could just sort it in the test. Good catch @yeya24